### ISSUE TITLE
Suggestion: Rename "registerTransient" to "register"

### ISSUE THREAD

**Issue by nonplus** [MisterGoodcat/good-injector#1]
The most common IOC container operation is to register transient types.  It would be nice to have this default registration operation called simply `register(When, Then)` rather than `registerTransient(When, Then)` (you can keep `registerTransient` as a deprecated alias).

This would also make the method name consistent with the `registerFactory` method.

BTW, I really like this simple, no-nonsense and easy-to-use approach (compared to Inversify).

**Comment by MisterGoodcat** [comment]
Good point, I'll do that for the next release. Thank you for your feedback!
